kernel/src/snapshot.rs

```diff
@@ -385,6 +400,48 @@ impl Snapshot {
 
         domain_metadata_configuration(self.log_segment(), domain, engine)
     }
+
+    /// Get the In-Commit Timestamp for this snapshot, if available.
+    /// It returns the inCommitTimestamp from the CommitInfo action of the commit that created this snapshot.
+    ///
+    /// Returns:
+    /// - `Ok(Some(timestamp))` if ICT is enabled and available for this snapshot
+    /// - `Ok(None)` if ICT is not enabled for this table, or if this snapshot predates ICT enablement
+    /// - `Err(...)` if ICT is enabled but there was an error reading it
+    pub fn get_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<Option<i64>> {
```

Comment:
I'd initially proposad DeltaResult, but let's also just hide this by making it pub(crate). There's no reason why a connector should know about ICT or need to read it.
```suggestion
    pub(crate) fn get_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<Option<i64>> {
```